tests/components/wmspro/test_cover.py

```diff
@@ -0,0 +1,223 @@
+) -> None:
+    """Test that a cover entity is created and updated correctly."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+    assert len(mock_hub_ping.mock_calls) == 1
+    assert len(mock_hub_configuration_prod.mock_calls) == 1
+    assert len(mock_hub_status_prod_awning.mock_calls) == 2
+
+    entity = hass.states.get("cover.markise")
+    assert entity is not None
+    assert entity.name == "Markise"
+
+    await async_setup_component(hass, "homeassistant", {})
+    await hass.services.async_call(
+        "homeassistant", "update_entity", {"entity_id": entity.entity_id}, blocking=True
+    )
+
+    assert len(mock_hub_status_prod_awning.mock_calls) == 3
+
+
+async def test_cover_close_and_open(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_hub_ping: AsyncMock,
+    mock_hub_configuration_prod: AsyncMock,
+    mock_hub_status_prod_awning: AsyncMock,
+    mock_action_call: AsyncMock,
+) -> None:
+    """Test that a cover entity is opened and closed correctly."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+    assert len(mock_hub_ping.mock_calls) == 1
+    assert len(mock_hub_configuration_prod.mock_calls) == 1
+    assert len(mock_hub_status_prod_awning.mock_calls) >= 1
+
+    entity = hass.states.get("cover.markise")
+    assert entity is not None
+    assert entity.state == "open"
+    assert entity.attributes["current_position"] == 100
+
+    with patch(
+        "wmspro.destination.Destination.refresh",
+        return_value=True,
+    ):
+        before = len(mock_hub_status_prod_awning.mock_calls)
+
+        await async_setup_component(hass, "cover", {})
+        await hass.services.async_call(
+            "cover", "close_cover", {"entity_id": entity.entity_id}, blocking=True
+        )
+
+        entity = hass.states.get("cover.markise")
+        assert entity is not None
+        assert entity.state == "closed"
+        assert entity.attributes["current_position"] == 0
+        assert len(mock_hub_status_prod_awning.mock_calls) == before
+
+    with patch(
+        "wmspro.destination.Destination.refresh",
+        return_value=True,
+    ):
+        before = len(mock_hub_status_prod_awning.mock_calls)
+
+        await async_setup_component(hass, "cover", {})
+        await hass.services.async_call(
+            "cover", "open_cover", {"entity_id": entity.entity_id}, blocking=True
+        )
+
+        entity = hass.states.get("cover.markise")
+        assert entity is not None
+        assert entity.state == "open"
+        assert entity.attributes["current_position"] == 100
+        assert len(mock_hub_status_prod_awning.mock_calls) == before
```

Comment:
I am curious, why does it change automatically? We don't update the state directly right?

Reply:
This mock here sets the requested parameters within the action as the new internal state: https://github.com/home-assistant/core/blob/811b990fbc4dd13d9cb23ac1101c2eec2ab43af1/tests/components/wmspro/conftest.py#L95-L106